cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/CachingPodsObserver.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2021 OpsMx.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.cats.redis.cluster;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.netflix.spinnaker.cats.agent.Agent;
+import com.netflix.spinnaker.cats.cluster.NodeIdentity;
+import com.netflix.spinnaker.cats.cluster.ShardingFilter;
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import com.netflix.spinnaker.kork.jedis.RedisClientDelegate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CachingPodsObserver implements ShardingFilter, Runnable {
+
+  private static final Logger logger = LoggerFactory.getLogger(CachingPodsObserver.class);
+  private int podCount = 0;
+  private int podIndex = -1;
+  private static final String REPLICA_SSET_KEY = "clouddriver:caching:replicas";
+  private static final String CORE_PROVIDER =
+      "com.netflix.spinnaker.clouddriver.core.provider.CoreProvider";
```

Comment:
I think its better to have an import like [how it is in the sql implementation PR](https://github.com/spinnaker/clouddriver/pull/5295/files#diff-7f9b8acd3580f646c1ac3c9b9957a63aeed7f8022e4a2764698e0f4b03c96abdR25) as this can easily be missed if there is any future refactoring of this class. Also, it would ensure consistency between the two implementations.

Reply:
It was leading to circular dependency so I had to do it this way. clouddriver-core depends on cats-redis but not on cats-sql. Importing CoreProvider into redis needs clouddriver-core to be a dependent.